src/jit/simd.cpp

```diff
@@ -1055,20 +1055,32 @@ GenTree* Compiler::impSIMDPopStack(var_types type, bool expectAddr, CORINFO_CLAS
 
     bool isParam = false;
 
-    // If we have a ldobj of a SIMD local we need to transform it.
-    if (tree->OperGet() == GT_OBJ)
+    // If we are popping a struct type it must have a valid handle.
+    // If 'structHandle' is specifed:
+    // - If we have an existing 'OBJ' we will change its handle if it doesn't match.
+    //   This can happen when we have a retyping of a vector that doesn't translate to any
+    //   actual IR.
+    // - (If it's not an OBJ and it's used in a parameter context where it is required,
+    //   impNormStructVal will add one).
+    // If 'structHandle' is not specified it is NO_CLASS_HANDLE), then we will use the
+    // handle on the stack value (see ti.GetClassHandleForValueClass() below).
+    //
+    if (structHandle != NO_CLASS_HANDLE)
```

Comment:
Most of the callers of `impSIMDPopStack` do not pass a struct handle. With this change `OBJ(ADDR(LCL_VAR))` won't be reduced to `LCL_VAR` anymore. Doesn't this cause CQ regressions?

Reply:
I'm not sure I understand the concern. This code will kick in only if the caller passed a valid class handle, in which case we want to ensure that the `OBJ`, if already present, will match. This addresses cases where we treat a vector conversion as a no-op, but don't want the `impNormStructType` code to assert when it encounters a non-matching handle.

Reply:
> This code will kick in only if the caller passed a valid class handle

Yes, but most callers pass `NO_CLASS_HANDLE`. And before this change we had
```C++
        GenTree* addr = tree->gtOp.gtOp1;
        if ((addr->OperGet() == GT_ADDR) && isSIMDTypeLocal(addr->gtOp.gtOp1))
        {
            tree = addr->gtOp.gtOp1;
        }
```
removing the `OBJ` node. Now this code is only executing if a valid class handle is passed in.

Reply:
Ah, I see. I'll rework it.